### ISSUE TITLE
process_formdata on empty valuelist on StringField

### ISSUE THREAD

**Issue by gdoumenc** [wtforms/wtforms#291]
On `StringField` class the method `process_formdata` assign `data` to empty string if `valuelist` is empty.
I think it is a wrong behavior as if an object was given and also a formdata, the value is replaced if the field is not in the formdata. Also all other field classes don't assign value on empty valuelist.

**Comment by dracarysX** [comment]
@gdoumenc  I cannot agree morel. I think this is a terrible behavior.

**Comment by Gawen** [comment]
I think this is definitely not what is required. Here's an example.

```py

class Form(wtforms.Form):
    foo = WhateverField()
    bar = StringField(default = "default_value")

f = Form({'foo': whatever})
print(repr(f.data['bar']))  # prints "", should print "default_value"
...

**Comment by sorgloomer** [comment]
Today I found out this was the reason behind flask-admin clearing up my string fields when disabled. This is how I reproduced it:

```
import wtforms


class MultiDict:
  def __init__(self, data):
    self.data = data
  def getlist(self, key):
    return self.data[key]
  def __iter__(self):
    return self.data.__iter__()

class MyForm(wtforms.Form):
    foo = wtforms.IntegerField()
    bar = wtforms.StringField(default = "default_value")

class MyObj:
  def __init__(self, foo, bar):
    self.foo = foo
    self.bar = bar

f = MyForm(MultiDict({'foo': [42]}))
print(repr(f.data['foo']), ', should be 42')  # prints 42, good
print(repr(f.data['bar']), ", should be 'default_value'")  # prints '', bad

f = MyForm(MultiDict({}), obj=MyObj(foo=31, bar='custom_value'))
print(repr(f.data['foo']), ', should be 31')  # prints 31, good
print(repr(f.data['bar']), ", should be 'custom_value'")  # prints '', bad
```

**Comment by davidism** [comment]
Released 2.2.1 which adjusts this to only set the empty string when data is not None. In 3.0 this will just leave data as None to be consistent with the other fields, but I didn't want to disrupt too much before 3.0.
